### PR TITLE
Reward application flow updates

### PR DESCRIPTION
**Pull request by msadura**
### <samp>🤖 Generated by Copilot at 07c12fa</samp>

This pull request implements the reward applications feature, which allows users to apply or submit work for rewards, and replaces the bounties feature. It updates several components in the `components/[pageId]/DocumentPage` and `components/rewards/components/RewardApplications` directories to use new UI components, styles, and data props. It also removes unused or redundant code and adds a new `NewWorkButton` component.

### WHY

- UI updates
- navigation fixes 

Screenshot 2023-10-31 at 13 16 02

Screenshot 2023-10-31 at 13 16 16

### CODE REVIEW

File: components/common/PageDialog/PageDialog.tsx [hunk 1]
```diff
@@ -139,7 +139,9 @@ export function PageDialog(props: Props) {
       }
       onClose={close}
     >
-      {page && <DocumentPage page={page} savePage={savePage} refreshPage={refreshPage} readOnly={readOnlyPage} />}
+      {page && (
+        <DocumentPage page={page} savePage={savePage} refreshPage={refreshPage} readOnly={readOnlyPage} close={close} />
```
Comment (msadura) [inline components/common/PageDialog/PageDialog.tsx:143]: unfortunately we had no way of closing pagedialog programatically

File: components/common/ExpandableSection.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import { ExpandableSectionTitle } from 'components/common/ExpandableSectionTitle';
+
+type Props = {
+  title: string;
+  forceExpand?: boolean;
+  children: React.ReactNode;
+};
+
+export function ExpandableSection({ title, forceExpand, children }: Props) {
```
Comment (msadura) [inline components/common/ExpandableSection.tsx:12]: this code was being copy-pasted all over the cards so I decded to go with simpel reusable component

File: components/rewards/components/RewardProperties/RewardProperties.tsx [hunk 3]
```diff
@@ -83,8 +75,13 @@ export function RewardProperties(props: {
         useDebouncedInputs
         values={currentReward}
         onChange={applyRewardUpdates}
+        readOnly={readOnly}
       />
 
+      {!!currentReward?.id && showApplications && (
```
Comment (motechFR) [inline components/rewards/components/RewardProperties/RewardProperties.tsx:81]: Important production consideration; We want the applications to be below the document page:
- Reward properties
- Document page
- Applications / submissions

**Review by motechFR — COMMENTED** [review]
Looks good, just need to resolve the main merge issues and we can give it a spin in staging

File: components/rewards/components/RewardProperties/RewardProperties.tsx (hunk 3 above)
Comment (msadura) [reply components/rewards/components/RewardProperties/RewardProperties.tsx:81]: https://discord.com/channels/894960387743698944/1134055482311577653/1166393253755625552

idea for this changed a bit, applications table is not supposed to be under custom properties

**Pull request merged**